CMakeLists.txt

```diff
@@ -1,116 +1,210 @@
-
-if(BUILD_TESTS)
-	file(GLOB TEST_SOURCES tests/*.cpp)
-	file(GLOB FUZZ_SOURCES tests/fuzz_*.cpp)
-	list(REMOVE_ITEM TEST_SOURCES ${FUZZ_SOURCES})
-
-	add_executable(check ${TEST_SOURCES})
-	if(BUILD_SHARED_AND_STATIC_LIBS)
-		target_link_libraries(check pugixml-static)
-	else()
-		target_link_libraries(check pugixml)
-	endif()
-	add_custom_command(TARGET check POST_BUILD COMMAND check WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR})
+target_sources(pugixml-shared
+  PRIVATE
+    $<${build-shared}:${PROJECT_SOURCE_DIR}/scripts/pugixml_dll.rc>
+    ${PROJECT_SOURCE_DIR}/src/pugixml.cpp)
+target_sources(pugixml-static
+  PRIVATE
+    ${PROJECT_SOURCE_DIR}/src/pugixml.cpp)
+target_sources(pugixml
+  PRIVATE
+    $<${build-shared}:${PROJECT_SOURCE_DIR}/scripts/pugixml_dll.rc>
+    ${PROJECT_SOURCE_DIR}/src/pugixml.cpp)
+
+# XXX: INSTALL_INTERFACE is not used here so that INCLUDES DESTINATION can work
+# correctly if USE_VERSIONED_LIBDIR is set. $<INSTALL_INTERFACE> can't expand
+# generator expressions otherwise.
+target_include_directories(pugixml-shared
+  PUBLIC
+    $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/src>)
+target_include_directories(pugixml-static
+  PUBLIC
+    $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/src>)
+target_include_directories(pugixml
+  PUBLIC
+    $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/src>)
+
+target_compile_definitions(pugixml-shared
+  PRIVATE
+    $<$<CXX_COMPILER_ID:MSVC>:PUGIXML_API=__declspec\(dllexport\)>)
+target_compile_definitions(pugixml
+  PRIVATE
+    $<$<CXX_COMPILER_ID:MSVC>:PUGIXML_API=__declspec\(dllexport\)>)
+
+target_compile_options(pugixml-shared
+  PRIVATE
+    ${msvc-rt-mtd-shared}
+    ${msvc-rt-mtd-static}
+    ${msvc-rt-mt-shared}
+    ${msvc-rt-mt-static})
+target_compile_options(pugixml
+  PRIVATE
+    ${msvc-rt-mtd-shared}
+    ${msvc-rt-mtd-static}
+    ${msvc-rt-mt-shared}
+    ${msvc-rt-mt-static})
+
+configure_package_config_file(
+  "${PROJECT_SOURCE_DIR}/scripts/pugixml-config.cmake.in"
+  "${PROJECT_BINARY_DIR}/pugixml-config.cmake"
+  INSTALL_DESTINATION ${CMAKE_INSTALL_DATADIR}
+  NO_CHECK_REQUIRED_COMPONENTS_MACRO
+  NO_SET_AND_CHECK_MACRO)
+
+write_basic_package_version_file(
+  "${PROJECT_BINARY_DIR}/pugixml-config-version.cmake"
+  COMPATIBILITY SameMajorVersion)
+
+configure_file(scripts/pugixml.pc.in pugixml.pc @ONLY)
+
+install(TARGETS ${install-targets}
+  EXPORT pugixml-targets
+  RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+  LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+  ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+  INCLUDES DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}${versioned-dir})
+
+install(EXPORT pugixml-targets
+  NAMESPACE pugixml::
+  DESTINATION ${CMAKE_INSTALL_DATADIR}/cmake/pugixml)
+
+install(FILES
+  "${PROJECT_BINARY_DIR}/pugixml-config-version.cmake"
+  "${PROJECT_BINARY_DIR}/pugixml-config.cmake"
+  DESTINATION ${CMAKE_INSTALL_DATADIR}/cmake/pugixml)
+
+install(FILES ${PROJECT_BINARY_DIR}/pugixml.pc
+  DESTINATION ${CMAKE_INSTALL_DATADIR}/pkgconfig)
```

Comment:
This seems wrong, the datadir is in /usr/share, whereas this used to be installed in /usr/lib.

Since it contains the libdir path and points to a .so, this needs to be installed to the libdir, not to the datadir.

This means that on Arch Linux (which I care about) this would be broken if we tried to create a lib32-pugixml package for multilib support.

Debian, which has native multi-arch, will definitely care about i386 and amd64 versions of the package installing the same file with different contents in /usr/share.

Reply:
ahh damn. That's a copy paste error. My mistake. I've been editing like 4 projects worth of CMake files. (TBQH I had figured no one was writing lib32/lib64 combinations anymore, but that's what I get for my hubris)

I can submit another PR with this fixed.